### PR TITLE
add information about RBAC bootstrap roles

### PR DESCRIPTION
**Pull request by deads2k**
Adds information about bootstrap RBAC roles, what they're for, and how to bind them.

@kubernetes/sig-auth-misc

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/2169)

### CODE REVIEW

File: docs/admin/authorization.md [hunk 1]
```diff
@@ -441,6 +441,55 @@ subjects:
   name: system:serviceaccounts
 ```
 
+### `system:*` prefixed RBAC resources
+
+If an RBAC resource (`ClusterRole`, `ClusterRoleBinding`, etc) is prefixed with `system:*`, that indicates that
+the resource is "owned" by the infrastructure.  Modifications to these resources can result in non-functional clusters.
+
+Once example is the `clusterrole/system:nodes`.  This role is used to provide limited permissions to kubelets.  If the
```
Comment (ncdc) [inline docs/admin/authorization.md]: One example is `clusterrole/system:nodes`

**Comment by erictune** [comment]
Thanks, looks good to me aside from ncdc's comment.

File: docs/admin/authorization.md [hunk 2]
```diff
@@ -441,6 +441,55 @@ subjects:
...
+One example is the `clusterrole/system:nodes`.  This role is used to provide limited permissions to kubelets.  If the
+role is modified, its possible to prevent your kubelet from working.
+
+### Default ClusterRoles and ClusterRoleBindings
+
+When starting up an API server without any ClusterRoles or ClusterRoleBindings, the API server will bootstrap itself
+with a set of default ClusterRoles and ClusterRoleBindings.  Most of these are `system:*` prefixed, but some are not.
+They are all labeled with `kubernetes.io/bootstrapping=rbac-defaults`.  These are the most commonly used:
+
+|Default Role |Description
```
Comment (timstclair) [inline docs/admin/authorization.md]: Table doesn't render correctly.  I think you're missing a line of `| --- | --- |`

File: docs/admin/authorization.md [hunk 3]
```diff
@@ -441,6 +441,55 @@ subjects:
...
+
+|*admin* |A project manager. If used in a `RoleBinding`, an *admin* user will have
```
Comment (seh) [inline docs/admin/authorization.md]: What is a "project"? Is it manifested as a Kubernetes namespace? Is it the Kubernetes cluster?

File: docs/admin/authorization.md [hunk 4]
```diff
@@ -441,6 +441,55 @@ subjects:
...
+|*cluster-admin* |A super-user role that can perform any action in any project. When
+granted to a user within a local policy, they have full control over quota and
+roles and every action on every resource in the project.
+
+|*cluster-status* |A role that can get basic cluster status information.
+
+|*edit* |A role that can modify most objects in a project, but does not have the
+power to view or modify roles or bindings.
+
+|*view* |A role who cannot make any modifications, but can see most objects in a
```
Comment (seh) [inline docs/admin/authorization.md]: s/who/that/, for consistency with line 468 above.

File: docs/admin/authorization.md [hunk 5]
```diff
@@ -441,6 +441,55 @@ subjects:
...
+project. They cannot view or modify roles or bindings. They cannot view secrets,
+since those are escalating.
+
+|*system:auth-delegator*|A role which allows delegated authentication and authorization
```
Comment (seh) [inline docs/admin/authorization.md]: s/which/that/, for consistency with line 468 above.

File: docs/admin/authorization.md [hunk 6]
```diff
@@ -441,6 +441,55 @@ subjects:
...
+checks.  This is commonly used by add-on API servers for a unified authentication and
+authorization experience.
+
+|*system:basic-user* |A role that can get basic information about himself.
+
+|*system:discovery*|A role which provides just enough power to access discovery and
```
Comment (seh) [inline docs/admin/authorization.md]: s/which/that/, for consistency with line 468 above.

File: docs/admin/authorization.md [hunk 7]
```diff
@@ -441,6 +441,55 @@ subjects:
   name: system:serviceaccounts
 ```
 
+### `system:*` prefixed RBAC resources
+
+If an RBAC resource (`ClusterRole`, `ClusterRoleBinding`, etc) is prefixed with `system:*`, that indicates that
```
Comment (seh) [inline docs/admin/authorization.md]: Add a period after "etc".

File: docs/admin/authorization.md [hunk 8]
```diff
@@ -441,6 +441,55 @@ subjects:
...
+the resource is "owned" by the infrastructure.  Modifications to these resources can result in non-functional clusters.
```
Comment (seh) [inline docs/admin/authorization.md]: s/non-functional/nonfunctional/

**Comment by deads2k** [comment]
@devin-donnelly @alex-mohr @cjcullen I'm referring people to this pull a lot and we really need a spot where we can merge this and then continue a guide.  Can we get a 1.6 branch or something so we can actively work on the docs in this area?

**Comment by liggitt** [comment]
@jaredbhatti any word on the 1.6 branch?

**Comment by chenopis** [comment]
@deads2k @liggitt FYI, the kubernetes:release-1.6 branch is now available. Please change the base branch to that. I will then be able to merge the PR. Thx

**Comment by liggitt** [comment]
included in https://github.com/kubernetes/kubernetes.github.io/pull/2360

**Comment by vish123al** [comment]
i am getting this error . can anyone plz help me
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {
    
  },
  "status": "Failure",
  "message": "services \"https:kubernetes-dashboard:\" is forbidden: User \"system:anonymous\" cannot get services/proxy in the namespace \"kube-system\"",
  "reason": "Forbidden",
  "details": {
    "name": "https:kubernetes-dashboard:",
    "kind": "services"
  },
  "code": 403
}

**Pull request closed without merging**